charts/presto/templates/deployment-coordinator.yaml

```
@@ -1,3 +1,4 @@
+{{- $podAnnotations := mergeOverwrite .Values.coordinator.podAnnotations .Values.podAnnotations }}
```

```suggestion
{{- $podAnnotations := mergeOverwrite .Values.podAnnotations .Values.coordinator.podAnnotations }}
```
The `mergeOverwrite` function merges dictionaries into one, giving precedence from right to left, see [docs](https://helm.sh/docs/chart_template_guide/function_list/#mergeoverwrite-mustmergeoverwrite).

I think that the value of a specific common annotation should be overwritten by the component-specific value.
For instance, the following values:
```yaml
podAnnotations:
  annotation-source-common: "true"
  check-merge-overwrite: "common"
coordinator:
  podAnnotations:
    annotation-source-coordinator: "true"
    check-merge-overwrite: "coordinator"
```
should render `coordinator` for the `check-merge-overwrite` value instead of `common`, i.e.
```yaml

spec:
  template:
    metadata:
      annotations:
        ...
        annotation-source-common: "true"
        annotation-source-coordinator: "true"
        check-merge-overwrite: coordinator
```

---

this is fixed now.